hardware_interface/package.xml

```diff
@@ -16,7 +17,8 @@
   <author>Adolfo Rodriguez Tsouroukdissian</author>
 
   <buildtool_depend>catkin</buildtool_depend>
+
   <depend>roscpp</depend>
-  <test_depend>rostest</test_depend>
+
   <test_depend>rosunit</test_depend>
```

Comment:
I'm wondering what rosunit brings us at this stage... gmock is integrated into catkin

Reply:
I took a look. I think it is safe to remove, https://github.com/ros/ros/tree/kinetic-devel/tools/rosunit

Reply:
[This doc](https://github.com/ros/catkin/blob/kinetic-devel/doc/howto/format2/gtest_configuration.rst) seems to indicate that it's still required, but that doc also looks horribly out of date (It looks to just be cloned over from package format 1, and untouched since 2014).

Ignoring that out-of-date doc, I think you're right. Thanks for looking into it.

I also did a quick sanity check against some other packages in the `ros` GitHub org. There aren't any `catkin_add_gmock`, but they have a lot of `catkin_add_gtest` (which is defined in [the same file](https://github.com/ros/catkin/blob/kinetic-devel/cmake/test/gtest.cmake)).

It's quite inconsistent (Likely because of legacy, I didn't dig through each package), but it appears the dependency is not needed nor standard.

I'm comfortable removing it if you are as well. Perhaps I should open up a PR to update that doc I linked above...

| Package | `catkin_add_gtest` | `find_package(rosunit)` | `<depend>rosunit</depend>` (package.xml only, excludes manifest.xml) |
| --- | :---: | :---: | :---: |
| actionlib |  ✔️ | ✔️ | ❌ |
| allocators |  ✔️ | ❌ | ❌ |
| angles |  ✔️ | ❌ | ✔️ |
| class_loader |  ✔️ | ❌ | ❌ |
| collada_urdf |  ✔️ | ❌ | ❌ |
| filters |  ✔️ | ❌ | ❌ |
| lockfree |  ✔️ | ❌ | ❌ |
| message_filters |  ✔️ | ✔️ | ✔️ |
| plugin_lib |  ✔️ | ❌ | ❌ |
| resource_retriever |  ✔️ | ❌ | ❌ |
| rosatomic |  ✔️ | ❌ | ❌ |
| rosbag_image_compressor |  ✔️ | ❌ | ❌ |
| rosbag_storage |  ✔️ | ❌ | ❌ |
| rosconsole |  ✔️ | ✔️ | ✔️ |
| roslaunch | ❌ (Uses `catkin_add_nosetests`) | ❌ | ✔️ |
| roslib |  ✔️ | ❌ | ❌ |
| roslz4 |  ✔️ | ❌ | ✔️ |
| rospack |  ✔️ | ❌ | ❌ |
| rosrt |  ✔️ | ✔️ | ✔️ |
| rostest |  ✔️ | ✔️ | ✔️ |
| rostime |  ✔️ | ❌ | ❌ |
| sensor_msgs |  ✔️ | ❌ | ✔️ |
| test_nodelet |  ✔️ | ❌ | ❌ |
| test_rosbag |  ✔️ | ✔️ | ✔️ |
| test_rosbag_storage |  ✔️ | ❌ | ❌ |
| test_roscpp |  ✔️ | ✔️ | ✔️ |
| test_rosservice | ❌ (Uses `catkin_add_nosetests`) | ✔️ | ✔️ |
| test_rospy | ❌ (Uses `catkin_add_nosetests`) | ✔️ | ❌ |
| test_tf2 |  ✔️ | ❌ | ✔️ |
| tf |  ✔️ | ❌ | ✔️ |
| tf_conversions |  ✔️ | ❌ | ❌ |
| tf2 |  ✔️ | ❌ | ❌ |
| tf2_bullet |  ✔️ | ❌ | ❌ |
| tf2_eigen |  ✔️ | ❌ | ❌ |
| tf2_geometry_msgs |  ✔️ | ❌ | ❌ |
| tf2_py |  ✔️ | ❌ | ❌ |
| topic_tools |  ✔️ | ✔️ | ✔️ |
| urdf |  ✔️ | ❌ | ❌ |
| xmlrpcpp |  ✔️ | ❌ | ❌ |

Reply:
Fixed in 37d1288e

Reply:
`catkin_lint` now complains that we're missing a `test_depend` on `rosunit`.

Perhaps we should get some outside help to clarify best practices here.

Reply:
Let's make a separate issue for this, this PR has reached a state where I'm eager to merge already ;)

Reply:
Sounds good. Would you rather leave it as is, with no `rosunit` dependency and `catkin_lint` complaints, or would you prefer I re-add the dep for now?

Reply:
Follow up in #408.

I'm pretty sure we're good without `rosunit` and that `catkin_lint` is just out of date (The rule for `catkin_add_gtest` hasn't been updated in 5 years), so I'm leaning towards leaving the code as is for this PR since I believe we likely have the correct configuration.

Reply:
If the CI passes, we should be ok.